### PR TITLE
[chore] address feedback for onboarding

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Changed text of language option from "N/A" to "Not Applicable"
- Removed upper number limit for time availability (hours per month)
- Added a character limit of 400 characters for availability description
- Added new radio field for legal credentials section to indicate whether the user is barred
- Added barred status comments for legal credential section to express additional information about barred status
    - Optional for users with bar number, required for users without

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @kyrenetam at the bottom."

No bar number
image

Has bar number
image

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Go in any order

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @varortz

### CODE REVIEW

File: src/app/onboarding/legal-credentials/page.tsx [hunk 1]
```diff
@@ -173,6 +215,35 @@ export default function Page() {
...
+                    onChange={newValue => {
+                      onboarding.updateProfile({
+                        legal_credential_comment: newValue,
+                      });
+                      field.onChange(newValue);
+                    }}
+                  />
+                </FormControl>
+                <FormDescription>
+                  For example, if you were formerly barred but is not currently.
```
Comment (varortz) [inline src/app/onboarding/legal-credentials/page.tsx]: nit: can we update this to "... are not currently."

File: src/components/SettingsSection/index.tsx [hunk 2]
```diff
@@ -54,7 +55,12 @@ export function SettingSection({
 
         {isEditing ? (
           <Flex $gap="1.25rem" $justify="end">
-            <Button type="button" onClick={() => cancelEdit?.()}>
+            <Button
+              type="button"
+              $secondaryColor={COLORS.redDark}
+              $tertiaryColor={COLORS.redDarker}
```
Comment (varortz) [inline src/components/SettingsSection/index.tsx]: Let's update these such that the `$secondaryColor` is `redMid` and `$tertiaryColor` is `redDark`. This way, it can be more consistent with the color of the Sign Out button: 
image

File: src/data/formSchemas.ts [hunk 3]
```diff
@@ -73,29 +73,48 @@ export const availabilitySchema = z.object({
...
+      })
+      .min(1, { message: 'Please include a state' }),
+    barred: z.boolean(),
+    barNumber: z
+      .string({ required_error: 'Please include your attorney bar number' })
+      .min(1, { message: 'Please include your attorney bar number' }),
+    eoirRegistered: z.boolean({ required_error: 'Must select one option' }),
+    legalCredentialComment: z
+      .string()
+      .max(400, 'Please keep it within 400 characters')
```
Comment (varortz) [inline src/data/formSchemas.ts]: Thoughts on rewording this to "Your text exceeds the 400-character limit."

File: src/data/formSchemas.ts [hunk 4]
```diff
@@ -135,6 +154,12 @@ export const roleAndLegalSchema = z
       })
       .optional()
       .nullable(),
+    barred: z.boolean().optional().nullable(),
```
Comment (varortz) [inline src/data/formSchemas.ts:155]: Could we update the wording here as well? ("Your text exceeds the 400-character limit.")

File: src/data/formSchemas.ts [hunk 5]
```diff
@@ -73,29 +73,48 @@ export const availabilitySchema = z.object({
...
   startDate: z
     .date({
       required_error:
         'Please include your estimated starting date of availability',
     })
     .min(getCurrentDate(), { message: 'Must select a current or future date' }),
-  availability: z.string().optional().nullable(),
+  availability: z
+    .string()
+    .max(400, 'Please keep it within 400 characters')
```
Comment (varortz) [inline src/data/formSchemas.ts]: Let's update the error message to "Your text exceeds the 400-character limit."?

File: src/app/onboarding/legal-credentials/page.tsx [hunk 6]
```diff
@@ -45,14 +46,21 @@ export default function Page() {
       stateBarred: onboarding.profile.state_barred ?? undefined,
       barNumber: onboarding.profile.bar_number ?? undefined,
       eoirRegistered: onboarding.profile.eoir_registered ?? undefined,
+      legalCredentialComment:
+        onboarding.profile.legal_credential_comment ?? undefined,
+      barred:
+        onboarding.profile.bar_number === undefined
+          ? undefined
+          : onboarding.profile.bar_number !== 'Not Barred',
```
Comment (varortz) [inline src/app/onboarding/legal-credentials/page.tsx]: can we update this to "No Bar Number Available". Apparently, there are some states, like New York, that do not have bar numbers.

File: src/app/onboarding/availability/page.tsx [hunk 7]
```diff
@@ -31,6 +31,7 @@ export default function Page() {
   const onboarding = useGuardedOnboarding();
   const { backlinkHref, ebbTo, pageProgress } = useOnboardingNavigation();
   const { push } = useRouter();
+  const [availabilityError, setAvailabilityError] = useState('');
```
Comment (varortz) [inline src/app/onboarding/availability/page.tsx:34]: Could we also provide instant feedback for the hours/month input on this page?

**Review by varortz — COMMENTED** [review]
Hey Jinkang, huge thank you for making these updates based on the feedback we received! Just added a couple comments, and I think it should be good to merge after we've addressed these :) Also, here are some additional things I noticed while reviewing:

## Onboarding Review Page:

- **Availability Section Overflow** - could we add `overflow-wrap: break-word;` to this text (similar to how we did with the summary for listings) to address this edge case:
Availability

- **Roles Section** - 
     - It seems like for attorneys, we're missing the additional information field. Could we go ahead and include this while also making sure it has the `overflow-wrap: break-word;` property?
     - Could we also go ahead and update "Legal Experience" to "Legal Credentials" for both attorneys and legal fellows.
     Legal Experience


## Settings
- Same as above, could we add the `overflow-wrap: break-word;` property to both the Roles and Availability section in the settings page?
Attorney-Specific
Availability

File: src/app/onboarding/legal-credentials/page.tsx (hunk 6 above)
Comment (jinkang-0) [reply src/app/onboarding/legal-credentials/page.tsx]: i'll add a separate boolean in supabase to indicate it, since the user can technically input "Not Barred" in the bar number field
and we can add any message on retool depending on the boolean; in supabase, it'll just be 'N/A'

File: src/app/onboarding/availability/page.tsx [hunk 8]
```diff
@@ -119,12 +120,21 @@ export default function Page() {
...
                         return;
                       }
 
                       const toNum = z.coerce.number().safeParse(newValue);
                       const num = toNum.success ? toNum.data : undefined;
 
+                      if (num === undefined) {
+                        setHoursError('Hours per month must be a number.');
+                      } else if (num < 0) {
+                        setHoursError('Hours per month cannot be negative.');
```
Comment (varortz) [inline src/app/onboarding/availability/page.tsx:133]: can we remove the periods at the end to stay consistent with the design?

File: src/data/formSchemas.ts [hunk 9]
```diff
@@ -1,6 +1,8 @@
 import z from 'zod';
 import { getCurrentDate } from '@/utils/helpers';
 
+export const CHAR_LIMIT_MSG = 'Your text exceeds the 400-character limit.';
+
```
Comment (varortz) [inline src/data/formSchemas.ts]: Same here, let's go ahead and remove the period

File: src/types/schema.d.ts [hunk 10]
```diff
@@ -28,6 +29,7 @@ export interface ProfileToUpload
     | 'eoir_registered'
     | 'bar_number'
     | 'legal_credential_comment'
+    | 'has_bar_number'
```
Comment (varortz) [inline src/types/schema.d.ts:32]: ty for adding this to the schema + supabase!

**Review by varortz — APPROVED** [review]
Looks great!!! Just left some a couple more comments on small fixes. Should be good to merge after they're addressed. 🕺
